python/tk_nuke_writenode/handler.py

```diff
@@ -996,6 +998,41 @@ def __set_profile(self, node, profile_name, reset_all_settings=False):
         self.__update_knob_value(node, "tk_file_type", file_type)
         self.__update_knob_value(node, "tk_file_type_settings", pickle.dumps(file_settings))
 
+        # Hide the promoted knobs that might exist from the previously
+        # active profile.
+        for promoted_knob in self._promoted_knobs:
+            promoted_knob.setFlag(nuke.INVISIBLE)
+        self._promoted_knobs = []
+        write_node = node.node(TankWriteNodeHandler.WRITE_NODE_NAME)
+        # We'll use link knobs to tie our top-level knob to the write node's
+        # knob that we want to promote.
+        i = 0
+        for knob_name in promote_write_knobs:
```

Comment:
Just a nicety for the enumeration, instead of managing ```i``` separately, something like this feels more pythonic:

```python
for (i, promoted_knob) in enumerate(self._promoted_knobs):
```